### DISCUSSION TITLE
Using DynamicDialog (useDialog composable) in component library

### DISCUSSION THREAD

**Discussion by nattyluke** [primefaces/community#307] [PrimeVue]
Hello,
first of all thank you for this wonderful library.
I created a custom Vue3 Component library based on primevue. Its working perfect.
But now I'm using DynamicDialogs (using useDialog composable). 
In my library test server it works fine, but when is use the library in my real app, it says:
Uncaught (in promise) Error: No PrimeVue Dialog provided!
(Even though using DynamicDialogs directly works without problem)
Somehow the library cant find the dialog service from the app, does anyone have a solution to this problem?

**Comment by tugcekucukoglu** [comment]
Hello,

Thanks for your kind words!

May I ask, did you add DialogService from main file to your app? For example;

```
import {createApp} from 'vue';
import DialogService from 'primevue/dialogservice';

const app = createApp(App);
app.use(DialogService);
```

**Comment by darwishdev** [comment]
i had the same issue but after so many struggles i found that my mistake was that 
i used this code to import the dialog at one of my components on my library
```ts
import { useDialog } from "primevue/usedialog";

```

and from another component i used this 
```ts
import { useDialog } from "primevue";

```

once i made them consistent it works well 

i guess this is because the code of useDialog on primevue as follows
```ts
import { inject } from "vue";

var PrimeVueDialogSymbol = Symbol();
function useDialog() {
  var PrimeVueDialog = inject(PrimeVueDialogSymbol);
  if (!PrimeVueDialog) {
    throw new Error("No PrimeVue Dialog provided!");
  }
  return PrimeVueDialog;
}

export { PrimeVueDialogSymbol, useDialog };
//# sourceMappingURL=index.mjs.map
```

you can see that they used new Symbol to generate a key for the injected service and this key is imported from the dialogService to provide the same key 

on my case the problem was that when i used to different import paths each one will get a new unique Symbol and that make the error happens 

no idea why this is happening only on library mode to be honest but eventually this fixed my issue

**Comment by trpl2alex14** [comment]
[resolve.dedupe](https://vite.dev/config/shared-options#resolve-dedupe) fixed my issue.

vite.config.js of my app:
```
export default defineConfig({
    ...
    resolve: {
        dedupe: [
	    'primevue'
        ],
    }
 })
```

it also solves this [problem](https://github.com/primefaces/primevue/issues/8126).
